src/main/java/org/wso2/carbon/module/swiftiso20022/utils/ValidatorUtils.java

```diff
@@ -18,21 +18,65 @@
 
 package org.wso2.carbon.module.swiftiso20022.utils;
 
+import org.wso2.carbon.module.swiftiso20022.constants.ConnectorConstants;
 import org.wso2.carbon.module.swiftiso20022.model.ErrorModel;
 
+import java.util.Currency;
+import java.util.Set;
+import java.util.regex.Pattern;
+
 /**
  * Class to validate the request payload.
  */
 public class ValidatorUtils {
 
-    /**
-     * Method to construct the error model.
+    /** Method to validate whether currency is in ISO 4217 format
+     *
+     * @param currency  Currency to be validated
+     * @return     Whether currency is valid
+     */
+    public static boolean isValidCurrency(String currency) {
+        if (currency == null || currency.isBlank()) {
+            return false;
+        }
+        Set<Currency> availableCurrencies = Currency.getAvailableCurrencies();
+        long matchingCurrencyCount = availableCurrencies.stream()
+                .filter(c -> c.getCurrencyCode().equals(currency)).count();
+        return matchingCurrencyCount > 0;
```

Comment:
Shall we consider using `anyMatch` here?

Reply:
Fixed in https://github.com/wso2-extensions/mediation-swift-iso20022-module/pull/2/commits/8d6b87e8988955fbe9679e7afa9d4285b0ff0ac8